### PR TITLE
Vite

### PR DESCRIPTION
**Pull request by lfarrell**
https://jira.lib.unc.edu/browse/BXC-3470

Also fixes numerous security issues and gets us around https://snyk.io/blog/peacenotwar-malicious-npm-node-ipc-package-vulnerability/

### CODE REVIEW

File: static/js/vue-cdr-access/src/components/collectionBrowse.vue [hunk 1]
```diff
@@ -8,7 +8,7 @@ Top level component used for the collection browse page
 </template>
 
 <script>
-    import listDisplay from "./listDisplay";
+    import listDisplay from "./listDisplay.vue";
```
Comment (bbpennel) [inline static/js/vue-cdr-access/src/components/collectionBrowse.vue:11]: why do some of these imports need to be switch to `'@/components/browseDisplay.vue';` form, but others can be left like this?

File: static/js/vue-cdr-access/src/components/searchWrapper.vue [hunk 2]
```diff
@@ -88,6 +88,9 @@ Top level component wrapper for search pages
             hasFacets() {
                 return this.facet_list.map(f => f.values).flat().length > 0;
             },
+            stuff(){
```
Comment (bbpennel) [inline static/js/vue-cdr-access/src/components/searchWrapper.vue:91]: Do we need the stuff()?

File: static/js/vue-cdr-access/src/mixins/routeUtils.js [hunk 3]
```diff
@@ -109,6 +109,16 @@ export default {
...
             return (error.name !== 'NavigationDuplicated' && !/^avoided\s+redundant\s+navigation/i.test(error.message));
+        },
+
+        /**
+         * Vite really, really wants all images to be in the project being built and referenced via imports.
+         * This doesn't make sense for some of our images. The project won't build unless an import or url is given.
+         * So just return the image url for images that are external to the Vue project.
+         * @param image
+         */
+        nonVueStaticImageUrl(image) {
```
Comment (bbpennel) [inline static/js/vue-cdr-access/src/mixins/routeUtils.js:120]: I'm only seeing this being used in one place, do you think it'd make sense to copy `ajax-loader-lg.gif` over into the project to make vue happy?

File: static/js/vue-cdr-access/tests/unit/viewType.spec.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import { shallowMount, flushPromises } from '@vue/test-utils'
 import { createRouter, createWebHistory } from 'vue-router';
-import viewType from '@/components/viewType.vue'
-import displayWrapper from "@/components/displayWrapper";
+import viewType from '../../src/components/viewType.vue'
```
Comment (bbpennel) [inline static/js/vue-cdr-access/tests/unit/viewType.spec.js:3]: Why do these two need to be changed to `../..` when it doesn't seem to be needed anywhere else?

File: static/js/admin/vue-permissions-editor/src/components/patronRoles.vue [hunk 5]
```diff
@@ -23,15 +23,15 @@
         <ul class="set-patron-roles">
             <li v-if="isBulkMode">
                 <input type="radio" v-model="user_type" value="ignore"
-                       id="user_type_ignore"> <label for="user_type_ignore"> No Change</label>
+                       id="user_type_ignore"> <label for="user_type_ignore" @click="setRoleFromLabel('ignore')"> No Change</label>
```
Comment (bbpennel) [inline static/js/admin/vue-permissions-editor/src/components/patronRoles.vue:26]: why is this near click binding and method needed now? Is it replacing any existing functionality?

File: static/js/vue-cdr-access/src/components/searchWrapper.vue (hunk 2 above)
Comment (lfarrell) [reply static/js/vue-cdr-access/src/components/searchWrapper.vue:91]: Hah! Forgot that was there.

File: static/js/vue-cdr-access/src/components/collectionBrowse.vue (hunk 1 above)
Comment (lfarrell) [reply static/js/vue-cdr-access/src/components/collectionBrowse.vue:11]: I think I meant to switch them all. Sometimes files couldn't be found if they weren't using the @/components syntax.

File: static/js/vue-cdr-access/src/mixins/routeUtils.js (hunk 3 above)
Comment (lfarrell) [reply static/js/vue-cdr-access/src/mixins/routeUtils.js:120]: I'm inclined to leave it. We'd have to move over an additional copy of the image during deploys and/or adjust the the image reference in the compiled js.

**Review by bbpennel — APPROVED** [review]

**Pull request merged**